api/versions.go

```diff
@@ -0,0 +1,67 @@
+// Copyright 2017 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package api
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"sort"
+
+	"github.com/web-platform-tests/wpt.fyi/shared"
+	"google.golang.org/appengine/datastore"
+)
+
+// apiVersionsHandler is responsible for emitting just the browser versions for the test runs.
+func apiVersionsHandler(w http.ResponseWriter, r *http.Request) {
+	product, err := shared.ParseProductParam(r)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	} else if product == nil {
+		http.Error(w, fmt.Sprintf("Invalid product param: %s", r.URL.Query().Get("product")), http.StatusBadRequest)
+		return
+	}
+
+	ctx := shared.NewAppEngineContext(r)
+	query := datastore.NewQuery("TestRun").Filter("BrowserName =", product.BrowserName)
+	queries := []*datastore.Query{query}
+	if product.BrowserVersion != "" {
+		queries = make([]*datastore.Query, 2)
+		queries[0] = query.Filter("BrowserVersion =", product.BrowserVersion).Limit(1)
+		queries[1] = shared.VersionPrefix(query, "BrowserVersion", product.BrowserVersion, false).
+			Project("BrowserVersion").
+			Distinct()
+	}
+
+	var runs shared.TestRuns
```

Comment:
This will initialize to `nil`. Is that safe for the first `append()`, or should we have `runs := make(shared.TestRuns, 0)`?

Reply:
`nil` arrays and empty arrays are equal (`len(foo) == 0`, etc)

Reply:
You're right that append nil-of-type-TestRuns will do what you expect (which is what matters here).

Aside: TIL, though, that an uninitialized TestRuns and make(TestRuns, 0) have the same length, but only the former is equal to nil.